src/lint/collect-header-diagnostics.ts

```diff
@@ -0,0 +1,104 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { getLocation } from './utils';
+
+const ruleId = 'header-format';
+
+export function collectHeaderDiagnostics(
+  dom: any,
+  headers: { element: Element; contents: string }[]
+) {
+  let lintingErrors: LintingError[] = [];
+
+  for (let { element, contents } of headers) {
+    if (!/\(.*\)$/.test(contents) || / Operator \( `[^`]+` \)$/.test(contents)) {
+      continue;
+    }
+
+    function indexToLineAndColumn(index: number) {
+      let headerLines = contents.split('\n');
+      let headerLine = 0;
+      let seen = 0;
+      while (true) {
+        if (seen + headerLines[headerLine].length >= index) {
+          break;
+        }
+        seen += headerLines[headerLine].length + 1; // +1 for the '\n'
+        ++headerLine;
+      }
+      let headerColumn = index - seen;
+
+      let elementLoc = getLocation(dom, element);
+      let line = elementLoc.startTag.line + headerLine;
+      let column =
+        headerLine === 0
+          ? elementLoc.startTag.col +
+            (elementLoc.startTag.endOffset - elementLoc.startTag.startOffset) +
+            headerColumn
+          : headerColumn + 1;
+
+      return { line, column };
+    }
+
+    let name = contents.substring(0, contents.indexOf('('));
+    let params = contents.substring(contents.indexOf('(') + 1, contents.length - 1);
+
+    if (/ $/.test(name)) {
+      name = name.substring(0, name.length - 1);
+    } else {
+      let { line, column } = indexToLineAndColumn(name.length);
+      lintingErrors.push({
+        ruleId,
+        nodeType: 'H1',
+        line,
+        column,
+        message: 'expected header to have a space before the argument list',
+      });
+    }
+
+    let nameMatches = [
+      /^(Runtime|Static) Semantics: [A-Z][A-Za-z0-9/]*$/,
+      /^(Number|BigInt)::[a-z][A-Za-z0-9]*$/,
```

Comment:
I don't think we need to restrict this to `Number`/`BigInt`.

```suggestion
      /^[A-Z][A-Za-z0-9]*::[a-z][A-Za-z0-9]*$/,
```